File: frontend/src/types/streaks.ts
export interface StreakData {
  habit_id: string
  current_streak: number
  longest_streak: number
  total_completions: number
  last_completed_date: string | null
}

export interface AllStreaksResponse {
  streaks: StreakData[]
}

export interface DayHistory {
  date: string
  completion_count: number
  total_habits: number
  completion_percentage: number
  has_journal_entry: boolean
  completed_habit_ids: string[]
  // Mood data (if journal entry has been analysed)
  mood_sentiment?: string | null
  mood_summary?: string | null
}

export interface HistoryResponse {
  days: DayHistory[]
}

File: backend/schemas/streaks.py
from pydantic import BaseModel
from datetime import date
from typing import Optional


class StreakResponse(BaseModel):
    habit_id: str
    current_streak: int
    longest_streak: int
    total_completions: int
    last_completed_date: Optional[date]


class AllStreaksResponse(BaseModel):
    streaks: list[StreakResponse]


class DayHistory(BaseModel):
    date: date
    completion_count: int
    total_habits: int
    completion_percentage: float
    has_journal_entry: bool
    completed_habit_ids: list[str]
    mood_sentiment: Optional[str] = None
    mood_summary: Optional[str] = None


class HistoryResponse(BaseModel):
    days: list[DayHistory]
